spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsSendingMessageHandler.java

```diff
@@ -70,6 +70,7 @@ public void setDestinationExpression(Expression destinationExpression) {
 		Assert.isTrue(this.destination == null && this.destinationName == null,
 				"The 'destination', 'destinationName', and 'destinationExpression' properties are mutually exclusive.");
 		this.destinationExpressionProcessor = new ExpressionEvaluatingMessageProcessor<Object>(destinationExpression);
+		setPrimaryExpression(destinationExpression);
```

Comment:
Yes, that was my first thought, but then I flipped to thinking the release strategy was more informational, or the output expression - hence my indecision 😄